cunumeric/deferred.py

```diff
@@ -442,26 +637,39 @@ def set_item(self, key, rhs):
         # Check to see if this is advanced indexing or not
         if is_advanced_indexing(key):
             # Create the indexing array
-            index_array = self._create_indexing_array(key)
-            if index_array.shape != rhs.shape:
-                raise ValueError(
-                    "Advanced indexing array does not match source shape"
-                )
-            if self.ndim != index_array.ndim:
-                raise NotImplementedError(
-                    "need support for indirect partitioning"
+            copy_needed, lhs, index_array = self._create_indexing_array(
+                key, True
+            )
+            # TODO: remove rhs.ndim ==0 logic when issue with scalars not being
+            # type of Store is addressed
+            if rhs.ndim == 0:
+                rhs_tmp = self.runtime.create_empty_thunk(
+                    index_array.base.shape,
+                    self.dtype,
+                    inputs=[],
                 )
+                task = self.context.create_task(CuNumericOpCode.FILL)
+                task.add_output(rhs_tmp.base)
+                task.add_input(rhs.base)
+                task.add_scalar_arg(False, bool)
+                task.execute()
+                rhs = rhs_tmp.base
+            else:
+                if rhs.shape != index_array.shape:
+                    rhs_tmp = rhs._broadcast(index_array.base.shape)
+                    rhs_tmp, rhs = rhs._copy_store(rhs_tmp)
+                else:
+                    rhs = rhs.base
```

Comment:
Don't you need to remove transformations from `rhs` in this case?

Reply:
All transformations should be removed in the `_create_indexing_array` call (or at `copy` call on line 660)

Reply:
I understand now. Yes, we need to remove all possible transformations here as well